### PR TITLE
Improve/Install Multiple plugins at once

### PR DESCRIPTION
**Pull request by tempodat**
* [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [x] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's devel branch if it's a completely
    new feature, or maintenance if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against master or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of master, maintenance or devel please),
    e.g. dev/my_new_feature or fix/my_bugfix
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [x] If your changes include style sheets: You have modified the
    .less source files, not the .css files (those are generated with
    lessc)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [x] You have run the existing unit tests against your changes and
    nothing broke
  * [x] You have added yourself to the AUTHORS.md file :)

#### What does this PR do and why is it necessary?
This PR Adds functionality to the plugin manager.
Allows selecting and installing/reinstalling multiple plugins at once in the plugin repository window.

#### How was it tested? How can it be tested by the reviewer?
Passed pytest and pre-commit. 

#### Any background context you want to provide?
Had to configure three octopi instances over the last month, was really frustrated with plugins installing one by one, and locking up the whole UI while at it.

#### Screenshots (if appropriate)
image
image
image

### CODE REVIEW

**Comment by foosel** [comment]
FYI, I've seen this, I have it in the backlog, I just have my hands full even more than usual this month, so it'll be a while until I come around to reviewing this.

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (cp2004) [inline package-lock.json:1]: This package-lock.json file shouldn't be here

**Review by foosel — CHANGES_REQUESTED** [review]
As already mentioned by @cp2004, there's a `package-lock.json` in there that doesn't belong, please remove it.

General notes:

  * `multiInstall` is more self documenting than `MI` as a prefix (which in my head btw also translates to "Monkey Island" and is thus very confusing albeit highly amusing). Please use that instead.
  * There's a lot of code duplication going on between `performMIJob` and `installPlugin`. The logic should only live in `installPlugin` and instead be utilized by the multi steps.
  * Throw the install tasks into a list and pop from that so you don't have to keep track of the current job.

**Comment by foosel** [comment]
Hey @tempodat, have you had a chance to look at the review comments yet? 🙂

**Pull request closed without merging**